src/mono/mono/mini/intrinsics.c

```diff
@@ -2138,7 +2138,25 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 	if (in_corlib &&
 		((!strcmp ("System.Numerics", cmethod_klass_name_space) && !strcmp ("Vector", cmethod_klass_name)) ||
 		!strncmp ("System.Runtime.Intrinsics", cmethod_klass_name_space, 25))) {
-		if (!strcmp (cmethod->name, "get_IsHardwareAccelerated")) {
+		const char* cmethod_name = cmethod->name;
+
+		if (strncmp(cmethod_name, "System.Runtime.Intrinsics.ISimdVector<System.Runtime.Intrinsics.Vector", 70) == 0) {
```

Comment:
if memory serves, MonoMethod->name is just the method name, not the fullname, so these tests won't work. i think you would want to do mono_method_get_name_full, but that allocates so it would slow down interp code transforms to a probably unacceptable degree. looking at `m_class_get_name(cmethod->klass)` plus the method name might be enough. i'm not sure where the generic args go in the name, if anywhere.

Reply:
This is checking just the method name., the full name includes the prefix `System.Runtime.Intrinsics.Vector128<T>.`

For C#, explicitly implemented intrinsics have names in the format of `FullNameOfInterface.Name`. For example see: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoKgOwEMBbGAZwAc6wYACAWQE8AcoxbtOVAN5Uu0roQDMXbDQwwoAMw7cAkvy3LVGsZRldJxkzMIpeACgCUAbikyAvlWfT5s4lwDCXEC4dPj0VdU0PU0iTLytbO1MuN3MLLmjLa2DQg00AOh57ROSTZJcgA

Note that we have both:
```
.method public hidebysig 
        instance void M () cil managed 
```

and
```
    .method private final hidebysig newslot virtual 
        instance void MyNamespace.IMyInterface.M () cil managed 
```

Reply:
-- so the method name is `MyNamespace.IMyInterface.M` and the full name of the method is `MyNamespace.C::MyNamespace.IMyInterface.M` (where I used `::` to separate the class name from the method name)

Reply:
I see, so in this specific case due to how we implement the interface methods, the name resembles a fullname